### PR TITLE
feat: extend teifilter model to include attribute value filters [DHIS2-12376]

### PR DESCRIPTION
**Pull request by ameenhere**
This PR extends the trackedentityinstancefilter table to add a jsonb column called "attributevaluefilters". This stores any filters related to tracked entity attribute values.

### CODE REVIEW

File: dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.38/V2_38_30__Add_attributefilter_column_to_trackedentityinstancefilter.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+alter table trackedentityinstancefilter add column if not exists attributevaluefilters jsonb default '[]'::jsonb;
```
Comment (enricocolasante) [inline dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.38/V2_38_30__Add_attributefilter_column_to_trackedentityinstancefilter.sql:1]: Duplicated script?
Or you just want to be really sure? :D

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/TrackedEntityInstanceFilterController.java [hunk 2]
```diff
@@ -38,7 +44,34 @@
  */
 @Controller
 @RequestMapping( value = TrackedEntityInstanceFilterSchemaDescriptor.API_ENDPOINT )
+@ApiVersion( include = { DhisApiVersion.ALL, DhisApiVersion.DEFAULT } )
```
Comment (enricocolasante) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/TrackedEntityInstanceFilterController.java:47]: Do we really need this annotation here?

**Review by enricocolasante — CHANGES_REQUESTED** [review]
Duplicated flyway script

File: dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.38/V2_38_30__Add_attributefilter_column_to_trackedentityinstancefilter.sql (hunk 1 above)
Comment (ameenhere) [reply dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.38/V2_38_30__Add_attributefilter_column_to_trackedentityinstancefilter.sql:1]: haha. Actually Zubair "reserved" the 2.38.29 flyway version so I renamed my file. but for some reason, it got pushed as a new file. Deleted the 2.38.29. The new script is 2.38.30 :) Good catch!

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/TrackedEntityInstanceFilterController.java (hunk 2 above)
Comment (ameenhere) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/TrackedEntityInstanceFilterController.java:47]: If the annotation is missing, api/{number}/rest/Of/the/url will not work. So I just made it to be standard. this means even if the api is accesed with or without the apiVersion, it continues to reach here.

**Review by enricocolasante — APPROVED** [review]

**Review by zubaira — APPROVED** [review]

**Pull request merged**